src/Symfony/Component/Dotenv/Dotenv.php

```diff
@@ -405,16 +433,24 @@ private function createFormatException($message)
         return new FormatException($message, new FormatExceptionContext($this->data, $this->path, $this->lineno, $this->cursor));
     }
 
-    private function doLoad(bool $overrideExistingVars, string $path, array $paths): void
+    /**
+     * @param bool $envMode Silently ignore not existing files in $paths (but not the one in $path)
+     */
+    private function doLoad(bool $overrideExistingVars, string $path, array $paths, bool $envMode = false): void
```

Comment:
`$envMode` variable name does not correspond with what it does. Should be named `$silent` (as it's used internally) or `$ignorePathErrors` or something like that.

PS. phpDoc is missing the other arguments

Reply:
$silent or $ignorePathErrors is less accurate: an error will be thrown of the main file (usually .env) doesn’t exist. Only files for specific environments are optional.

Regarding the PHPDoc, in Symfony we don’t add the useless lines.

Reply:
IMO `$envMode` is not meaningful. If you don't look at implementation, only on signature, would you know what `$envMode` does? Honestly, it doesn't even look like `bool` variable, rather string or bitwise flag.

Reply:
@Wirone I agree, it's why I added a PHPDoc description :) If you have better name in mind, I'll be very happy to switch (`$silent` and `$ignorePathErrors` were my first thoughts, but they are more confusing than `$envMode`).

Reply:
What about `$safe` (determines if call is safe and will not throw an exception) or `$required` (determines if all paths should exist - it would require changing to `$silent = !$required`).

Reply:
IMO they are not more explicit, and can be confusing. `environment mode` is exactly what is is, and `$ignoreFilesThatNotExistExceptTheFirstOne` is a bit long. Isn't the description in the PHPDoc enough? This method is private, it will never be used by the end user, only by Symfony contributors (we may expect that they read the PHPDoc :D).

Reply:
`doLoad(bool $overrideExistingVars, string $path, bool $ignoreMissingExtraPaths, string ...$extraPaths = array())`?

Reply:
Not sure that using the splat operator is a good idea in this private method (we may add extra parameters at some point), also it is already used without the splat operator in other functions.

Reply:
parameter renamed